### PR TITLE
Deprecate broadcast_zpreserving[!]

### PR DESCRIPTION
**Pull request by Sacha0**
This pull request deprecates `Base.SparseArrays.broadcast_zpreserving[!]`, please see #19533 for more. Best!

### CODE REVIEW

File: base/sparse/sparsematrix.jl [hunk 1]
```diff
@@ -2281,6 +2170,10 @@ floor{To}(::Type{To}, A::SparseMatrixCSC) = floor.(To, A)
 trunc{To}(::Type{To}, A::SparseMatrixCSC) = trunc.(To, A)
 round{To}(::Type{To}, A::SparseMatrixCSC) = round.(To, A)
 
+# broadcast_zpreserving[!] deprecated in 0.6. Wasn't exported prior, so need these
+# definitions for the deprecation to serve its purpose. Remove after 0.6.
```
Comment (tkelman) [inline base/sparse/sparsematrix.jl]: could also eval the deprecation into this module from deprecated.jl

Comment (Sacha0) [inline base/sparse/sparsematrix.jl]: Much cleaner. Done. Thanks!

**Comment by stevengj** [comment]
My inclination is just to do a straight `@deprecate _broadcast_zpreserving(f, args...) broadcast(f, args...)` here.   I can't find a single package actually using this function; if someone needs it, and for some reason `broadcast` doesn't do what they want, they can file an issue.

**Comment by Sacha0** [comment]
> My inclination is just to do a straight @deprecate _broadcast_zpreserving(f, args...) broadcast(f, args...) here.

Being gentle with the deprecation costs us nothing at this point, the work already being done, so might as well keep the softer deprecation? Thanks!

**Comment by stevengj** [comment]
Sure

**Comment by Sacha0** [comment]
Absent objections or requests for time, I plan to merge this PR Thursday morning PDT. Best!

**Comment by Sacha0** [comment]
Thanks for reviewing / resolving conflicts and merging!

**Pull request merged**